core/core-impl/src/main/java/ai/timefold/solver/core/config/heuristic/selector/move/composite/UnionMoveSelectorConfig.java

```diff
@@ -162,6 +166,31 @@ public void visitReferencedClasses(Consumer<Class<?>> classVisitor) {
         classVisitor.accept(selectorProbabilityWeightFactoryClass);
     }
 
+    @Override
+    public UnionMoveSelectorConfig enableNearbySelection(Class<? extends NearbyDistanceMeter<?, ?>> distanceMeter,
+            Random random) {
+        UnionMoveSelectorConfig nearbyConfig = copyConfig();
+        List<MoveSelectorConfig> updatedMoveSelectorList = new LinkedList<>();
+        for (NearbyAutoConfigurationMoveSelectorConfig<?> selectorConfig : moveSelectorConfigList.stream()
+                .filter(s -> NearbyAutoConfigurationMoveSelectorConfig.class.isAssignableFrom(s.getClass()))
+                .map(s -> (NearbyAutoConfigurationMoveSelectorConfig<?>) s).toList()) {
```

Comment:
Id argue that the following would read better, also coincidentally doesn't need to create an extra list or iterate twice:

   for (... all configs ...) {
       if (not nearby) {
           continue;
       }
       ...
   }

Also, the `selectorConfig` variable is another nice use of `var` - all the necessary context is already there.